src/components/badge/badge.style.ts

```diff
@@ -36,14 +42,19 @@ const StyledBadge = styled.span.attrs(({ onClick }) => ({
   width: 22px;
   min-height: 22px;
   border: solid 2px transparent;
-  border-color: var(--colorsActionMajor500);
-  color: var(--colorsActionMajor500);
+
+  ${({ customColor, theme }) => css`
```

Comment:
**suggestion**: I find all the conditional checks on `customColor` in this component a bit ugly to read. I think we can avoid all that by making `customColor` a mandatory prop (for `StyledBadge` I mean, not for `Badge` itself! note that this will mean you can't use my earlier suggestion of `Pick<BadgeProps, "customColor">` if you do this), and then when rendering the styled component pass it as `customColor={customColor || "--colorsActionMajor500"}`.

This way we avoid a lot of the complexity in this component that's not really needed, and instead will just be saying "we'll render whatever color is passed in, but if one isn't it'll be `colorsActionMajor500`". Which I think is what we want?